### ISSUE TITLE
problem with nginx and bcrypt for htpasswd

### ISSUE THREAD

**Issue by Nightbr** [jwilder/nginx-proxy#643]
Hey,

I have an issue with htpasswd and bcrypt encryption. When I encrypt my password with bcrypt, I get a 500 error from nginx-proxy and this error:

2016/11/28 16:00:36 [crit] 16#16: *6090 crypt_r() failed (22: Invalid argument), client: 178.16.163.28, server: MYDOMAIN, request: "GET / HTTP/1.1", host: "MYDOMAIN", referrer: "MYDOMAIN"

It's linked to this issue https://github.com/docker/distribution/issues/655 that nginx doesn't support bcrypt.
I think you just need to update nginx image and it will work. For example, it works with `nginx:alpine`.

For now the workaround I used is to generate htpasswd using SSHA but it's not ultra secured...

NOTE: Generate htpasswd:

`docker run --rm httpd htpasswd -nb ${USER} ${PASSWORD} > auth.htpasswd`

**Comment by cdaringe** [comment]
yes, i also have this issue.  on my host i did:

```
  docker run --entrypoint htpasswd registry:2 -Bbn $UN $PW > /www/registry/auth/registry.$DOMAIN
  docker-compose up # which brings up nginx-proxy:latest and registry:2
```

the full message from the container is:

```
nginx_proxy        | nginx.1    | 2017/01/29 05:40:33 [crit] 19#19: *1 crypt_r() failed (22: Invalid argument), client: 73.25.92.242, server: registry.<my-domain>, request: "GET / HTTP/1.1", host: "registry.<my-domain>"
```

**Comment by cdaringe** [comment]
i also tried using `htpsswd` from within the nginx container to gen the pw. no dice

**Comment by Nightbr** [comment]
@cdaringe thank you ! Nice workaround using alpine version ;)

@jwilder you can close this issue if using the alpine version is ok for you. But I think it could be nice to add some documentation about it ;)

**Comment by toopay** [comment]
Try use  `-d` flag when generate the password. Force CRYPT encryption works for me.

**Comment by denyshubh** [comment]
Hi, I too was facing the same problem. I used

```
sudo sh -c "echo -n 'kibanaadmin:' >> /etc/nginx/htpasswd.users"
sudo sh -c "openssl passwd -apr1 >> /etc/nginx/htpasswd.users"
```
to setup up my password  and it worked for me.
You can refer this post if you are trying to setup kibana and got this issue.
https://medium.com/@shubham.singh98/log-monitoring-with-elk-stack-c5de72f0a822?postPublishedType=repub

**Comment by BernardZijlstra** [comment]
Bcrypt would be a very nice to have. For Nginx to be able to use bcrypt hashes as a much safer password storage for basic authentication instead of storing passwords in MD5 hashes. Please consider adding it to Nginx.

**Comment by hkmaly** [comment]
Yes ; It's nice when documentation mentions (in https://nginx.org/en/docs/http/ngx_http_auth_basic_module.html) that sha1 passwords are unsafe, but so are md5.
